src/structures/paging/frame.rs

```diff
@@ -132,6 +161,18 @@ impl<S: PageSize> Sub<PhysFrame<S>> for PhysFrame<S> {
     }
 }
 
+impl<S: PageSize> CheckedSub<PhysFrame<S>> for PhysFrame<S> {
+    type Output = u64;
+    #[inline]
+    fn checked_sub(self, rhs: PhysFrame<S>) -> Option<Self::Output> {
+        self.start_address()
+            .checked_sub(rhs.start_address())
+            .map(PhysAddr::new)
```

Comment:
I opted for `new`, which can panic over `try_new` as the sub should prevent ever falling into panic inducing address space unless someone is doing something really bananas. I can happily change this to `try_new` if you would prefer since `new` just calls `try_new` under the hood. This would guarantee we entirely avoid a panic that I _think_ won't ever even happen, but at the expense of denser code.

Reply:
Nevermind. The more i thought about this I opted to change it. The guarantee of no panic for a `checked_*` method should probably be the default.

Reply:
What is the point of converting the offset to a `PhysAddr` and `PhysFrame`? Couldn't you just write:
```rust
self.start_address()
    .checked_sub(rhs.start_address())
    .map(|offset| offset / S::SIZE)
```

Reply:
Hello, I opted for the case of safely validating of the address and the alignment as there are a few cases where an unaligned frame start address could be returned. This would necessitate the caller broke the contract of [from_start_address_unchecked](https://github.com/rust-osdev/x86_64/blob/master/src/structures/paging/frame.rs#L39). I'm happy to change this method if you feel strongly about it.

Reply:
Because `from_start_unchecked` is marked as unsafe, we can always assume that the caller did not violate the safety requirements and we don't need to check for them again. 

Even if the caller did violate the requirements, this would cause the function to fail as if an underflow occurred even when that's not the case, which seems a bit unexpected.

Reply:
Fair, the change you've requested has been made.

Reply:
I noticed that you also used this pattern in a few other places, those probably need to be changed too.

Reply:
Applied the same change to `Page::checked_sub` to match the same. By other's i assume you mean something like https://github.com/rust-osdev/x86_64/pull/298/files#diff-d2a15219d98ad66c2aeac0ee03d682034bc619fe5905e63960599aa58b7a63cfR332-R342, in which this was done for the sake of checking valid addresses of the multiple and I think there is no reason to change that.

Reply:
You don't need to check if the offset is a valid address, in fact it is wrong to do so. For addition it's perfectly fine to jump from the lower half of the address space to the upper half. In that case the offset is not necessarily a valid virtual address. Offsets just aren't the same as virtual (or physical) addresses.

I'm aware that there has been some discussion on whether we should allow jumping from the lower half to the upper half and I'm open to change in that direction. However for now we should make sure that the checked and non-checked implementations behave in the same way.